lib/DefineOpenCLWorkItemBuiltinsPass.cpp

```diff
@@ -382,11 +384,21 @@ bool DefineOpenCLWorkItemBuiltinsPass::defineGlobalOffsetBuiltin(Module &M) {
     auto Dim = &*F->arg_begin();
     auto InBoundsDim = inBoundsDimensionIndex(Builder, Dim);
     Value *Indices[] = {Builder.getInt32(0), InBoundsDim};
-    auto GoffPtr =
-        GetPushConstantPointer(BB, clspv::PushConstant::GlobalOffset);
-    auto GoffDimPtr = Builder.CreateInBoundsGEP(GoffPtr, Indices);
-    auto GoffDim = Builder.CreateLoad(GoffDimPtr);
-    auto Ret = inBoundsDimensionOrDefaultValue(Builder, Dim, GoffDim, 0);
+    Value *gep = nullptr;
+    const bool uses_push_constant = clspv::ShouldDeclareGlobalOffset(M);
```

Comment:
A little pedantic, but the function name `clspv::ShouldDeclareGlobalOffset()` is a little confusing now, since it specifically means "should declare global offset _as a push constant_". When I first read this line of code I thought it was a bug until I read the definition of that function.

Reply:
I'll clarify the name.